app/listenconfig_data.go

```diff
@@ -75,11 +76,36 @@ func addListenConfig(management *config.ManagementContext, cfg Config) error {
 	}
 
 	if existing == nil {
-		_, err := management.Management.ListenConfigs("").Create(cfg.ListenConfig)
+		if _, err := management.Management.ListenConfigs("").Create(cfg.ListenConfig); err != nil {
+			return err
+		}
+	} else {
+		cfg.ListenConfig.ResourceVersion = existing.ResourceVersion
+		if _, err := management.Management.ListenConfigs("").Update(cfg.ListenConfig); err != nil {
+			return err
+		}
+	}
+
+	if selfSigned {
+		data := map[string]string{}
+		data["tls.key"] = cfg.ListenConfig.CAKey
+		data["tls.crt"] = cfg.ListenConfig.CACert
+
+		secret := &corev1.Secret{
+			StringData: data,
+			Type:       corev1.SecretTypeTLS,
+		}
+		secret.Name = "tls-rancher"
+		secret.Namespace = "cattle-system"
+
+		if _, err := management.Core.Secrets("cattle-system").Get("tls-rancher", v1.GetOptions{}); apierrors.IsNotFound(err) {
+			_, err = management.Core.Secrets("cattle-system").Create(secret)
+			return err
+		}
+
+		_, err = management.Core.Secrets("cattle-system").Update(secret)
```

Comment:
is there a way to compare the content to eliminate updates when nothing had changed?

Reply:
checked that `selfSigned` is set true only when new cert is generated, thus also covering the case of no updates